src/wp-includes/rest-api/endpoints/class-wp-rest-pattern-directory-controller.php

```diff
@@ -80,6 +80,7 @@ public function get_items_permissions_check( $request ) {
 	 * Search and retrieve block patterns metadata
 	 *
 	 * @since 5.8.0
+	 * @since 6.0.0 Added 'slug' to request.
```

Comment:
In the Gutenberg plugin, this class extends `WP_REST_Pattern_Directory_Controller`:

https://github.com/WordPress/gutenberg/blob/cab8cc52cf715722d8d8aa081ee66744e991ab27/lib/compat/wordpress-6.0/class-gutenberg-rest-pattern-directory-controller.php#L12

In the backport, however, it extends `WP_REST_Controller`. Is this intentional?

Reply:
Ah, I see that the Gutenberg version extends this one here, presumably to override some functionality which can be just merged right into the code in wordpress-develop.

Reply:
Good question @adamziel. 

The Gutenberg plugin extends `WP_REST_Pattern_Directory_Controller` to add functionality to the existing controller that was added in 5.8.0. The inheritance chain in the plugin is: `Gutenberg_REST_Pattern_Directory_Controller` extends `WP_REST_Pattern_Directory_Controller` extends `WP_REST_Controller`. 

This PR merges the changes from `Gutenberg_REST_Pattern_Directory_Controller` into `WP_REST_Pattern_Directory_Controller`.